src/sql/platform/connection/common/connectionConfig.ts

```diff
@@ -241,6 +241,11 @@ export class ConnectionConfig {
 				profile.id = generateUuid();
 				changed = true;
 			}
+			// SSMS 19 requires "user", fix any profiles created without user property.
+			if (profile.providerName === 'MSSQL' && profile.options.user === undefined) {
```

Comment:
I would use isUndefinedOrNull for any of these strict checks

https://github.com/microsoft/azuredatastudio/blob/08311728b9c240f1b3f290b5821e9ffd36301f7c/src/vs/base/common/types.ts#L83